source/hp/dof_handler.cc

```diff
@@ -1284,7 +1284,18 @@ namespace hp
     Assert (ff.size() > 0,
             ExcMessage("The hp::FECollection given is empty!"));
 
-    if (&fe_collection != &ff)
+    // don't create a new object if the one we have is already appropriate
+    bool is_same = (fe_collection.size() == ff.size());
```

Comment:
one could as well add `operator==` for `FECollection`, that could be useful elsewhere.